lib/core/src/rodsLog.cpp

```diff
@@ -69,6 +70,30 @@ std::string create_log_error_prefix() {
     return ret;
 }
 
+int rodsLog_derive_verbosity_level_from_legacy_log_level()
+{
+    if (CLIENT_PT == ::ProcessType) {
+        return verbosityLevel;
+    }
+
+    namespace log_ns = irods::experimental::log;
+
+    switch (log_ns::get_level_from_config(irods::KW_CFG_LOG_LEVEL_CATEGORY_LEGACY)) {
+        using log_ns::level;
+
+        case level::trace:      return LOG_SQL;
+        case level::debug:      return LOG_DEBUG7;
+        case level::info:       return LOG_NOTICE;
+        case level::warn:       return LOG_WARNING;
+        case level::error:      return LOG_ERROR;
+        case level::critical:   return LOG_SYS_FATAL;
+    }
+
+    // We should never reach this line because get_level_from_config() always returns
+    // a valid log level (i.e. it catches all exceptions and defaults to level::info).
+    return LOG_SQL;
```

Comment:
Would it make more sense to return `LOG_NOTICE` here, based on the comment?

Reply:
Okay. The comment kind of suggested that the default would result in `LOG_NOTICE`, which is why I said that. Perhaps we can add to the comment to explain why `LOG_SQL` is returned instead? What you've written here is good enough explanation, IMO.

Reply:
Will do.

Reply:
Done.